### PR TITLE
Updates to Helidon Common, Part 2

### PR DESCRIPTION
**Pull request by tomas-langer**
This is a major update of Helidon common modules and refactors all modules that use them.
Most notable changes are in
1. `helidon-common-http` (headers) 
2. `helidon-common-media` - media type support including identifying media type based on file suffixes; now used in both HTTP and Config (and static content)

Commits separate this into smaller batches (still quite big) of changes related to modules as I was processing them.

### CODE REVIEW

File: common/http/pom.xml [hunk 1]
```diff
@@ -27,19 +27,28 @@
     <modelVersion>4.0.0</modelVersion>
     <artifactId>helidon-common-http</artifactId>
     <name>Helidon Common HTTP</name>
+    <description>Common HTTP primitives used by both blocking and reactive servers</description>
 
     <dependencies>
         <dependency>
             <groupId>io.helidon.common</groupId>
-            <artifactId>helidon-common-reactive</artifactId>
+            <artifactId>helidon-common</artifactId>
```
Comment (danielkec) [inline common/http/pom.xml:35]: image

**Review by romain-grecourt — APPROVED** [review]

**Review by danielkec — APPROVED** [review]
1/16 LGTM

**Review by barchetta — APPROVED** [review]

File: common/http/pom.xml (hunk 1 above)
Comment (tomas-langer) [reply common/http/pom.xml:35]: Common http is now used by both Helidon SE and new blocking server. Cannot have reactive in. Sorry

Comment (danielkec) [reply common/http/pom.xml:35]: The end of an era it is

**Pull request merged**